modules/database/src/ioc/db/dbConvertJSON.c

```diff
@@ -135,43 +109,42 @@ static int dbcj_start_array(void *ctx) {
     return (parser->depth == 1);
 }
 
-static int dbcj_end_array(void *ctx) {
-    parseContext *parser = (parseContext *) ctx;
-
-    parser->depth--;
-    return (parser->depth == 0);
-}
-
-
 static yajl_callbacks dbcj_callbacks = {
     dbcj_null, dbcj_boolean, dbcj_integer, dbcj_double, NULL, dbcj_string,
-    dbcj_start_map, dbcj_map_key, dbcj_end_map,
-    dbcj_start_array, dbcj_end_array
+    dbcj_start_map, NULL, NULL,
+    dbcj_start_array, NULL
 };
 
 long dbPutConvertJSON(const char *json, short dbrType,
     void *pdest, long *pnRequest)
 {
     parseContext context, *parser = &context;
-    yajl_alloc_funcs dbcj_alloc;
     yajl_handle yh;
     yajl_status ys;
     size_t jlen = strlen(json);
     long status;
 
-    if(INVALID_DB_REQ(dbrType))
+    if (INVALID_DB_REQ(dbrType)) {
+        errlogPrintf("dbConvertJSON: Invalid dbrType %d\n", dbrType);
         return S_db_badDbrtype;
+    }
+
+    if (!jlen) {
+        *pnRequest = 0;
+        return 0;
+    }
 
     parser->depth = 0;
     parser->dbrType = dbrType;
     parser->dbrSize = dbValueSize(dbrType);
     parser->pdest = pdest;
     parser->elems = *pnRequest;
 
-    yajl_set_default_alloc_funcs(&dbcj_alloc);
-    yh = yajl_alloc(&dbcj_callbacks, &dbcj_alloc, parser);
-    if (!yh)
+    yh = yajl_alloc(&dbcj_callbacks, NULL, parser);
```

Comment:
Passing NULL for alloc_funcs will call yajl_set_default_alloc_funcs() inside yajl_alloc() anyway.

Reply:
True, I think I only learned that after writing this code.